### DISCUSSION TITLE
Listing releases does not return latest release

### DISCUSSION THREAD

**Discussion by harpener** [community/community#67702] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hi guys,

in past weeks we have repeatedly encountered issues with missing releases for existing tags.

We are using [Github API for creating a release](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release). However, it sometimes starts failing due to existing tag. Release for such tag is not returned during [Github API for listing releases](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#list-releases), nor it is visible in Github UI, but it is visible using [Github API for latest release](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-the-latest-release). I haven't seen any error returned from Github API, the release seems to have been created successfully.

Example (private) repository:
https://github.com/roihunter/composer - tag name 1.0.155 (I might re-create it later to fix the issue)

Do you have any idea why is this happening?

Thank you for your answer and have a nice day.

**Comment by Sachin066** [comment]
Check for Tag Conflicts: Ensure that there are no conflicting tags in your repository. If a tag with the same name already exists for a different commit, it could lead to unexpected behavior when creating releases.

Use the Correct API Endpoint: Double-check that you are using the correct GitHub API endpoint for creating releases. The endpoint for creating a release is POST /repos/:owner/:repo/releases.

API Responses: You mentioned that the release seems to have been created successfully, but it's not visible in the GitHub UI or when listing releases via the API. Check the response you receive from the GitHub API when creating the release for any clues or error messages. Sometimes, API responses can provide insights into what might be happening.

**Reply by harpener** [reply]
> Check for Tag Conflicts: Ensure that there are no conflicting tags in your repository. If a tag with the same name already exists for a different commit, it could lead to unexpected behavior when creating releases.

Yes, this exact error we are getting because the tag exists, but the release is not returned and our automation is based on returned releases.

> Use the Correct API Endpoint: Double-check that you are using the correct GitHub API endpoint for creating releases. The endpoint for creating a release is POST /repos/:owner/:repo/releases.

We are using the correct API endpoints.

> API Responses: You mentioned that the release seems to have been created successfully, but it's not visible in the GitHub UI or when listing releases via the API. Check the response you receive from the GitHub API when creating the release for any clues or error messages. Sometimes, API responses can provide insights into what might be happening.

I might not have been clear when I said that releases seem created successfully, I meant that API response from Github API says exactly that and contains no errors whatsoever.

I should also point out that I cannot tell whether the release disappears in time or was never visible in the first place, because such issue is only detected on next release. And it does not happen consistently, only sometimes.

Thanks for the advices.
